diff --git a/Makefile b/Makefile
index bea0556..fde7199 100644
--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,15 @@
+ifeq ($(strip $(DESTDIR)),)
+	INSTALLBASE = $(HOME)/.local/share/gnome-shell/extensions
+else
+	INSTALLBASE = $(DESTDIR)/usr/share/gnome-shell/extensions
+endif
+
 all: build install
 
 .PHONY: build install lint dist
 
 build:
-	./update-locale.sh
+	sh ./update-locale.sh
 	glib-compile-schemas --strict --targetdir=caffeine@patapon.info/schemas/ caffeine@patapon.info/schemas
 
 dist: build
@@ -11,8 +17,8 @@ dist: build
 	cd caffeine@patapon.info && zip -r ../caffeine@patapon.info.zip ./* --exclude \*.po --exclude \*.gschema.xml
 
 install:
-	install -d ~/.local/share/gnome-shell/extensions
-	cp -a caffeine@patapon.info/ ~/.local/share/gnome-shell/extensions/
+	install -d $(INSTALLBASE)
+	cp -a caffeine@patapon.info/ $(INSTALLBASE)
 
 lint:
 	eslint --resolve-plugins-relative-to "$(shell npm root -g)" caffeine@patapon.info
